async.c

```diff
@@ -616,19 +621,40 @@ static int __redisAsyncCommand(redisAsyncContext *ac, redisCallbackFn *fn, void
     cb.pending_subs = 1;
 
     /* Find out which command will be appended. */
-    p = nextArgument(cmd,&cstr,&clen);
-    assert(p != NULL);
+    if (NULL == (p = nextArgument(cmd,&cstr,&clen))) {
+        __redisSetError(c, REDIS_ERR_PROTOCOL, "Invalid command format");
+        __redisAsyncCopyError(ac);
+        return REDIS_ERR;
+    }
+
     hasnext = (p[0] == '$');
     pvariant = (tolower(cstr[0]) == 'p') ? 1 : 0;
     cstr += pvariant;
     clen -= pvariant;
 
+    /* NOTICE:
+     *  Further non thread-safe actions [ we append to buffer before pushing having proper callback ]
+     *  Since hiredis is already not thread-safe let's just append first to check for OOM errors */
+
+    /* Make sure we can append command before making it's callback - Delegate error */
+    if (REDIS_OK != (ret = __redisAppendCommand(c,cmd,len))) return ret;
+
+#define SAFE_PUSHCALLBACK(cb_list) \
```

Comment:
I'm not a fan of the "SAFE" prefix. Just do `PUSH_CALLBACK`.

Reply:
Fixed